itest/lnd_sweep_test.go

```diff
@@ -171,52 +168,36 @@ func testSweepCPFPAnchorOutgoingTimeout(ht *lntest.HarnessTest) {
 	))
 	ht.MineEmptyBlocks(int(numBlocks))
 
-	// Assert Bob's force closing tx has been broadcast.
-	closeTxid := ht.AssertNumTxsInMempool(1)[0]
+	// Assert Bob's force closing tx has been broadcast. We should see two
+	// txns in the mempool:
+	// 1. Bob's force closing tx.
+	// 2. Bob's anchor sweeping tx CPFPing the force close tx.
+	_, sweepTx := ht.AssertForceCloseAndAnchorTxnsInMempool()
 
 	// Remember the force close height so we can calculate the deadline
 	// height.
 	forceCloseHeight := ht.CurrentHeight()
 
-	// Bob should have two pending sweeps,
-	// - anchor sweeping from his local commitment.
-	// - anchor sweeping from his remote commitment (invalid).
-	//
-	// TODO(yy): consider only sweeping the anchor from the local
-	// commitment. Previously we would sweep up to three versions of
-	// anchors because we don't know which one will be confirmed - if we
-	// only broadcast the local anchor sweeping, our peer can broadcast
-	// their commitment tx and replaces ours. With the new fee bumping, we
-	// should be safe to only sweep our local anchor since we RBF it on
-	// every new block, which destroys the remote's ability to pin us.
-	sweeps := ht.AssertNumPendingSweeps(bob, 2)
+	var anchorSweep *walletrpc.PendingSweep
 
-	// The two anchor sweeping should have the same deadline height.
-	deadlineHeight := forceCloseHeight + deadlineDeltaAnchor
-	require.Equal(ht, deadlineHeight, sweeps[0].DeadlineHeight)
-	require.Equal(ht, deadlineHeight, sweeps[1].DeadlineHeight)
-
-	// Remember the deadline height for the CPFP anchor.
-	anchorDeadline := sweeps[0].DeadlineHeight
+	// Bob should have one pending sweep,
+	// - anchor sweeping from his local commitment.
+	expectedNumSweeps := 1
 
-	// Mine a block so Bob's force closing tx stays in the mempool, which
-	// also triggers the CPFP anchor sweep.
-	ht.MineEmptyBlocks(1)
+	// For neutrino backend, Bob would have two anchor sweeps - one from
+	// the local and the other from the remote.
```

Comment:
I agree more documentations would be nice, and you are welcome to create PRs to add them. As for this PR, I don't think going through all the neutrino case is a trivial job, since they vary from case to case - some related to unconfirmed utxos, some related to no `testmempoolaccept` available, etc.